packages/jest-rut/src/integration.ts

```diff
@@ -9,11 +9,13 @@ export function runWithTimers<T>(cb: () => T): T {
   }
 
   const value = cb();
+  let loop = 0;
 
-  if (jest.getTimerCount() !== 0) {
+  while (jest.getTimerCount() > 0 && loop < 10) {
```

Comment:
would there be an edge case where a test has 11 timers/promises and this breaks?

Reply:
11 timers wouldn't be an issue, since the `run` methods exhaust *all* timers. This is to simply avoid an infinite loop, caused by intervals or something similar.